appservice/test/index.ts

```diff
@@ -3,64 +3,55 @@
  *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-//
-// PLEASE DO NOT MODIFY / DELETE UNLESS YOU KNOW WHAT YOU ARE DOING
-//
-// This file is providing the test runner to use when running extension tests.
-// By default the test runner in use is Mocha based.
-//
-// You can provide your own test runner if you want to override it by exporting
-// a function run(testRoot: string, clb: (error:Error) => void) that the extension
-// host can call to run the tests. The test runner is expected to use console.log
-// to report the results back to the caller. When the tests are finished, return
-// a possible error to the callback or null if none.
-
+import * as glob from 'glob';
+import * as Mocha from 'mocha';
 import * as path from 'path';
-// tslint:disable-next-line:no-require-imports no-submodule-imports
-import testRunner = require('vscode/lib/testrunner');
 
-const options: { [key: string]: string | boolean | number | object } = {
-    ui: 'tdd', 		// the TDD UI is being used in extension.test.ts (suite, test, etc.)
-    useColors: true // colored output from test results
-};
+// tslint:disable-next-line: export-name
+export async function run(): Promise<void> {
+    const options: Mocha.MochaOptions = {
+        ui: 'tdd',
+        color: true,
+        reporter: 'mocha-multi-reporters',
+        reporterOptions: {
+            reporterEnabled: 'spec, mocha-junit-reporter',
+            mochaJunitReporterReporterOptions: {
+                mochaFile: path.join(__dirname, '..', '..', 'test-results.xml')
+            }
+        }
+    };
+
+    addEnvVarsToMochaOptions(options);
+    console.log(`Mocha options: ${JSON.stringify(options, undefined, 2)}`);
```

Comment:
Also why 2?

Reply:
It's a formatting thing to make the output look nicer. 2 is the number of spaces to insert as tabs